jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpDateTime.java

```diff
@@ -0,0 +1,229 @@
+                {
+                    second = StringUtil.toInt(token, 6);
+                    minute = StringUtil.toInt(token, 3);
+                    hour = StringUtil.toInt(token, 0);
+                    continue;
+                }
+
+                // RFC 6265 - Section 5.1.1 - Step 2.2
+                if (day == (-1) && token.length() <= 2)
+                {
+                    day = StringUtil.toInt(token, 0);
+                    continue;
+                }
+
+                // RFC 6265 - Section 5.1.1 - Step 2.3
+                if (month == (-1) && token.length() == 3)
+                {
+                    Integer m = MONTH_CACHE.getBest(token);
+                    if (m != null)
+                    {
+                        month = m;
+                        continue;
+                    }
+                }
+
+                // RFC 6265 - Section 5.1.1 - Step 2.4
+                if (year == (-1))
+                {
+                    if (token.length() <= 2)
+                    {
+                        year = StringUtil.toInt(token, 0);
+                    }
+                    else if (token.length() == 4)
+                    {
+                        year = StringUtil.toInt(token, 0);
+                    }
+                    continue;
+                }
+            }
+        }
+        catch (Throwable x)
+        {
+            if (LOG.isDebugEnabled())
+                LOG.debug("Ignore: Unable to parse Date/Time", x);
+        }
+
+        // RFC 6265 - Section 5.1.1 - Step 3
+        if ((year > 70) && (year <= 99))
+            year += 1900;
+        // RFC 6265 - Section 5.1.1 - Step 4
+        if ((year >= 0) && (year <= 69))
+            year += 2000;
+
+        // RFC 6265 - Section 5.1.1 - Step 5
+        if (day == (-1))
+            throw new IllegalArgumentException("Missing [day]: " + datetime);
+        if (month == (-1))
+            throw new IllegalArgumentException("Missing [month]: " + datetime);
+        if (year == (-1))
+            throw new IllegalArgumentException("Missing [year]: " + datetime);
+        if (hour == (-1))
+            throw new IllegalArgumentException("Missing [time]: " + datetime);
+        if (day < 1 || day > 31)
+            throw new IllegalArgumentException("Invalid [day]: " + datetime);
+        if (month < 1 || month > 31)
+            throw new IllegalArgumentException("Invalid [month]: " + datetime);
+        if (hour > 23)
+            throw new IllegalArgumentException("Invalid [hour]: " + datetime);
+        if (minute > 59)
+            throw new IllegalArgumentException("Invalid [minute]: " + datetime);
+        if (second > 59)
+            throw new IllegalArgumentException("Invalid [second]: " + datetime);
+
+        // RFC 6265 - Section 5.1.1 - Step 6
+        ZonedDateTime dateTime = ZonedDateTime.of(year,
+            month, day, hour, minute, second, 0,
+            ZoneId.of("GMT"));
```

Comment:
This should be UTC not GMT and there is a constant available:
```suggestion
            ZoneOffset.UTC;
```

Reply:
UTC and GMT are different.

GMT is a Time Zone based on  
UTC is a time standard representing an offset origin based on atomic clocks.

GMT and UTC are not exactly aligned with time too.
In Java, UTC is represented as a 'Z', while GMT is `GMT`.

`ZoneOffset` is not a `ZoneID` and cannot be used interchangeably.

The `ZonedDateTime` tracks ZoneOffset and ZoneID separately.

Note: if you use a `ZoneOffset` in the date/time APIs you are often required to ALSO provide a `ZoneID`.

Reply:
I know UTC and GMT are different.  The RFC says to use UTC so that is primarily why I commented.  

`ZoneOffset` extends `ZoneID` so it is-a `ZoneID`  and the implementation of `ZoneID.getOffset(long)` is `{return this}`.  So `ZoneOffset.UTC` is a perfectly valid constant `ZoneID` to use.  Feel free to make your own constant for `ZoneId.of("UTC")`, but I think you will find it will just return the `ZoneOffset.UTC` instance anyway!